### PR TITLE
Add more unit tests for RemoteStoreUtils and RemoteFsTimestampAwareTranslog

### PR DESCRIPTION
**Pull request by sachinpkale**
### Description
- Add more unit tests for code flows that are missing test coverage.

### Check List
- [x] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java [hunk 1]
```diff
@@ -476,7 +476,7 @@ private static Set<String> getPinnedTimestampLockedFiles(
         Set<Long> newPinnedTimestamps = new TreeSet<>(Collections.reverseOrder());
         for (Long pinnedTimestamp : pinnedTimestampSet) {
             String cachedFile = metadataFilePinnedTimestampMap.get(pinnedTimestamp);
-            if (cachedFile != null) {
+            if (cachedFile != null && metadataFiles.contains(cachedFile)) {
```
Comment (sachinpkale) [inline server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java:479]: Extra check to make sure that returned list of locked metadata files will always be a sub-set of provided list of metadata files.

File: server/src/main/java/org/opensearch/index/translog/RemoteFsTimestampAwareTranslog.java [hunk 2]
```diff
@@ -434,15 +435,15 @@ protected static void deleteStaleRemotePrimaryTerms(
...
             logger.trace("No metadata is uploaded yet, returning from deleteStaleRemotePrimaryTerms");
             return;
         }
-        Optional<Long> minPrimaryTermFromMetadataFiles = metadataFiles.stream().map(file -> {
+        Optional<Long> minPrimaryTermFromMetadataFiles = metadataFilesNotToBeDeleted.stream().map(file -> {
             try {
                 return getMinMaxPrimaryTermFromMetadataFile(file, translogTransferManager, oldFormatMetadataFilePrimaryTermMap).v1();
             } catch (IOException e) {
-                return Long.MAX_VALUE;
+                return Long.MIN_VALUE;
```
Comment (sachinpkale) [inline server/src/main/java/org/opensearch/index/translog/RemoteFsTimestampAwareTranslog.java:446]: As we compare with `>` operation against `minPrimaryTermInRemote`, we need to use `Long.MIN_VALUE` as default value.

File: server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java (hunk 1 above)
Comment (sachinpkale) [reply server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java:479]: Makes sense. Let me add assert as well.

Comment (sachinpkale) [reply server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java:479]: @gbbafna Do you mean removing the check and adding it as an assert?

Comment (sachinpkale) [reply server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java]: Made the change

**Review by gbbafna — APPROVED** [review]

**Pull request merged**